### PR TITLE
SearchField documentation

### PR DESCRIPTION
**Pull request by ross-pfahler**
Closes RSP-1264

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      <div>Mirrored text: {currentText}</div>
+      <div>Submitted text: {submittedText}</div>
+    </div>
+  );
+}
+```
+
+## Props
+
+<PropTable component={docs.exports.SearchField} links={docs.links} />
```
Comment (ross-pfahler) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx]: `PropTable` doesnt appear correct in the examples, but im not sure why. anyone have ideas?

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+<SearchField aria-labelledby="label-1" marginStart="10px" />
+```
+
+### Internationalization
+
+In order to internationalize a SearchField, a localized string should be passed to the `label` or `aria-label` prop.
+When the `necessityIndicator` prop is set to `"label"`, a localized string will be provided for `"(required)"` or `"(optional)"` automatically.
+
+```tsx example
+<SearchField label="Street address" isRequired necessityIndicator="label" marginEnd="20px" />
```
Comment (ktabors) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx]: Should these be in a provider that changes the language to show a localized string? If so, should the label should be be a localized value as well?

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 1 above)
Comment (ktabors) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: Did you see any interesting console messages in the docs start?

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+and the contents can still be copied. See [the MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly) for more information.
+
+```tsx example
+<SearchField label="Email" defaultValue="abc@adobe.com" isReadOnly />
+```
+
+### Label Alignment
+
+Labels can be aligned to the "start" or the "end" of the SearchField. For left-to-right (LTR) languages, "start" refers to the left most edge
+of the SearchField and "end" refers to the right most edge. For right-to-left (RTL) languages, this is flipped.
```
Comment (ktabors) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx]: `this is flipped.` => `this is mirrored.`
I think in other docs we say mirrored, right?

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+```
+
+## Props
+
+<PropTable component={docs.exports.SearchField} links={docs.links} />
+
+## Visual Options
+
+### Quiet 
+
```
Comment (ktabors) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx]: It seems like we include a descriptive sentence when we don't have a link to the spectrum docs. This is an observation to start a discussion.

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 3 above)
Comment (ross-pfahler) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: i think we use both (i copied this from TextField). I'll grep and adjust.

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 1 above)
Comment (ross-pfahler) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: i do not.

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 2 above)
Comment (ross-pfahler) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: yes, see the first sentence. Or am i missing your point?

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+```
+
+## Events
+
+The most commonly used handlers for events in SearchField are the:
+- `onChange` prop which is triggered whenever the value is edited by the user. 
+- `onSubmit` prop which is triggered whenever the value is submitted by the user. 
+- `onClear` prop which is triggered whenever the value is cleared by the user. 
+
+The example below uses `onChange`, `onSubmit`, and `onClear` to update a separate elements with the text entered into the SearchField.
```
Comment (LFDanLu) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx]: "update a separate elements" -> "update two separate elements"

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 1 above)
Comment (LFDanLu) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: Ahhhhhh, did a bit of digging, turns out we don't export SpectrumSearchFieldProps lol (https://github.com/adobe-private/react-spectrum-v3/blob/master/packages/@react-types/searchfield/src/index.d.ts#L20). Toss a export in front and you should be good

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 2 above)
Comment (ktabors) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: I'm wondering about changing the example to be like https://github.com/adobe-private/react-spectrum-v3/pull/197/files#diff-ae417a798db001f2a7e68cdfe7a842a9R84-R95

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  packageData={packageData}
+  componentNames={['SearchField']} />
+
+## Example
+
+```tsx example
+<SearchField
+  label="Search"
+  placeholder="Enter text"
+  onSubmit={text => alert(`Search for ${text}`)} />
```
Comment (snowystinger) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx]: Any chance we could make this a less annoying example somehow? maybe write it out to a sibling span? i know it makes the example more complicated, but i dislike the alerts

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+Placeholder text will only appear when the SearchField is empty, and should not be used as a substitute for labeling the component with a visible label.
+
+```tsx example
+<SearchField placeholder="abc@adobe.com" label="Email" />
+```
+
+## Labeling
+
+A visual label should be provided for the SearchField using the `label` prop. If the SearchField is required, the `isRequired` and 
+`necessityIndicator` props can be used to show a required state.
```
Comment (snowystinger) [inline packages/@react-spectrum/searchfield/docs/SearchField.mdx]: I know search field has inherited necessityIndicator, but is that actually something a search field should ever display? Why would a search field ever be required for a form? it doesn't really provide data

Comment (ross-pfahler) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: if the form only has a search field and a submit button, then its basically required, no?

i'm happy to adjust if you think it should be adjusted.

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 6 above)
Comment (ross-pfahler) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: ok, done.

i think we should have a mock console or something below the examples.

File: packages/@react-spectrum/searchfield/docs/SearchField.mdx (hunk 7 above)
Comment (snowystinger) [reply packages/@react-spectrum/searchfield/docs/SearchField.mdx]: Yeah, this one is really more of a question/pondering on my part, haha. Why would a form have a search? or why would a required form have a search? in what scenario would a search be required?
I'll still approve, but now i'm curious, haha

**Pull request merged**